# message type that corresponds to shingle.py
Header header
int8 id
int8 x_coord
int8 y_coord
int8[] neighbors_ids
int8[] neighbors_status
bool on_frontier 
int8 edge 
bool is_half_shingle
int8 shingle_status